models/issues/issue_dev_link.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package issues
+
+import (
+	"context"
+	"strconv"
+
+	"code.gitea.io/gitea/models/db"
+	git_model "code.gitea.io/gitea/models/git"
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+type IssueDevLinkType int
+
+const (
+	IssueDevLinkTypeBranch IssueDevLinkType = iota + 1
+	IssueDevLinkTypePullRequest
+)
+
+type IssueDevLink struct {
+	ID            int64 `xorm:"pk autoincr"`
+	IssueID       int64 `xorm:"INDEX"`
+	LinkType      IssueDevLinkType
+	LinkedRepoID  int64                  `xorm:"INDEX"` // it can link to self repo or other repo
+	LinkIndex     string                 // branch name, pull request number or commit sha
```

Comment:
We have branch table, so this can be int64?
ps: Commit SHA seems meaningless

Reply:
In future PRs, we can link a commit to this issue. So it will be kept to support that. BTW: Azure supports linking to a commit. I think for some development model, they just push commits with no feature branches or pull requests, they can use this feature.

Reply:
Then it can be split into to columns. `LinkIndex`(int64) and `LinkRef`(string)?
Using `string` type means you need to convert it from/to `int64` type every time if it is a PR ID or branch ID which seems not good.

Reply:
> Then it can be split into to columns. `LinkIndex`(int64) and `LinkRef`(string)? Using `string` type means you need to convert it from/to `int64` type every time if it is a PR ID or branch ID which seems not good.

The conversion is on the memory and we don't know what data we will store according to different LinkType. And this column will never be used as index, so we don't need to consider performance or as search condition.